From fc5be29f9b2cd9fc7243bf3c26ed042f2868fa8b Mon Sep 17 00:00:00 2001
From: Kejia Hu <kejia.hu@codethink.co.uk>
Date: Tue, 28 Apr 2020 08:04:37 +0100
Subject: [PATCH] power: reset: qcom-pon: reg write mask depends on pon generation
Git-commit: fc5be29f9b2cd9fc7243bf3c26ed042f2868fa8b
Patch-mainline: v5.8-rc1
References: git-fixes

Instead of hardcode the mask, it should be depends
on which generation of pon it was.

Signed-off-by: Kejia Hu <kejia.hu@codethink.co.uk>
Fixes: fce5430f6a86 ("reset: qcom-pon: Add support for gen2 pon")
Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/power/reset/qcom-pon.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/reset/qcom-pon.c b/drivers/power/reset/qcom-pon.c
index 22a743a0bf28..4a688741a88a 100644
--- a/drivers/power/reset/qcom-pon.c
+++ b/drivers/power/reset/qcom-pon.c
@@ -34,7 +34,8 @@ static int pm8916_reboot_mode_write(struct reboot_mode_driver *reboot,
 
 	ret = regmap_update_bits(pon->regmap,
 				 pon->baseaddr + PON_SOFT_RB_SPARE,
-				 0xfc, magic << pon->reason_shift);
+				 GENMASK(7, pon->reason_shift),
+				 magic << pon->reason_shift);
 	if (ret < 0)
 		dev_err(pon->dev, "update reboot mode bits failed\n");
 
-- 
2.16.4

